lib/app/features/core/views/components/app_lifecycle_observer.dart

```diff
@@ -3,19 +3,16 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_hooks/flutter_hooks.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
-import 'package:ice/app/features/core/providers/permissions_provider.dart';
+import 'package:ice/app/features/core/permissions/providers/permissions_provider.dart';
 
-class LifecycleWatcher extends HookConsumerWidget {
-  const LifecycleWatcher({required this.child, super.key});
+class AppLifecycleObserver extends HookConsumerWidget {
+  const AppLifecycleObserver({required this.child, super.key});
 
   final Widget child;
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    useOnAppLifecycleStateChange((
-      _,
-      AppLifecycleState current,
-    ) {
+    useOnAppLifecycleStateChange((_, AppLifecycleState current) {
       if (current == AppLifecycleState.resumed) {
         ref.read(permissionsProvider.notifier).checkAllPermissions();
```

Comment:
And this check we also can move to the `permissionsProvider`.

Reply:
done